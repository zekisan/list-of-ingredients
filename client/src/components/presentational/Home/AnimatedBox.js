import React, { Component } from 'react';
import AnimateHeight from 'react-animate-height';

class AnimatedBox extends Component {
    constructor() {
        super();

        this.toggle = this.toggle.bind(this);
    }

    state = {
        height: 0,
    };

    toggle() {
        const { height } = this.state;

        this.setState({
            height: height === 0 ? 'auto' : 0,
        });
    };

    render() {
        const { height } = this.state;

        return (
            <div>
                <button onClick={this.toggle}>
                    {height === 0 ? 'Open' : 'Close'}
                </button>

                <AnimateHeight
                    duration={500}
                    height={height} // see props documentation bellow
                >
                    <h1>Your content goes here</h1>
                    <p>Put as many React or HTML components here.</p>
                </AnimateHeight>
            </div>
        );
    }
}

export default AnimatedBox;
