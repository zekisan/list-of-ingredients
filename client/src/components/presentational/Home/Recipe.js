import React from 'react';
import PropTypes from 'prop-types';
import { Container, Header, Segment, Button } from 'semantic-ui-react';

const Recipe = ({ drink }) => <span>
  {drink &&
    <Container>
      <Header as='h2'>{drink.title}</Header>
      {drink.description && <p>{drink.description}</p>}
      {drink.ingredients &&
        <Segment.Group>
          {drink.ingredients.map((ingredient, i) => <Segment key={i}>{ingredient.description}</Segment>)}
        </Segment.Group>
      }
      {drink.steps && <p>{drink.steps}</p>}
      {drink.source && <Button basic size='tiny' color='teal' href={drink.source}>Source</Button>}
    </Container>
  }
</span>;

Recipe.propTypes = {
  /** the drink data */
  drink: PropTypes.object,
};

export default Recipe;