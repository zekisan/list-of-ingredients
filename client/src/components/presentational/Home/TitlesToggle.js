import React from 'react';
import PropTypes from 'prop-types';
import { Container, Button } from 'semantic-ui-react';

const TitlesToggle = ({ drinks, drink, getDrink }) =>
  <span>
    {
      drinks && drinks.length
        ? <Button.Group color='teal' fluid widths={drinks.length}>
          {Object.keys(drinks).map((key) => {
            return <Button active={drink && drink.id === drinks[key].id} key={key} onClick={() => getDrink(drinks[key].id)}>
              {drinks[key].title}
            </Button>
          })}
        </Button.Group>
        : <Container textAlign='center'>No drinks found.</Container>
    }
  </span>;

TitlesToggle.propTypes = {
  drinks: PropTypes.array,
  drink: PropTypes.object,
  getDrink: PropTypes.func,
};

export default TitlesToggle;
