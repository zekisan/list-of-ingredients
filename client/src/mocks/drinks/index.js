const drinks = [
    {"id":1,"title":"Two-Minute Breakfast Boost","description":"Whizz up a low-fat breakfast smoothie in no time. Use banana with other soft fruit, plus honey for a little sweetness and oats for slow-release fuel.","steps":"Put all the ingredients in a blender and whizz for 1 min until smooth. Pour the mixture into two glasses to serve.","source":"https://www.bbcgoodfood.com/recipes/two-minute-breakfast-smoothie","created_at":"2018-06-09T00:22:22.524Z","updated_at":"2018-06-09T00:22:22.524Z","ingredients":[{"id":1,"description":"1 banana"},{"id":2,"description":"1 tbsp porridge oats"},{"id":3,"description":"80g soft fruit (like mango or strawberries)"},{"id":4,"description":"150ml milk"},{"id":5,"description":"1 tsp honey"},{"id":6,"description":"1 tsp vanilla extract"}]},
    {"id":2,"title":"Kale And Hearty Smoothie","description":"Give yourself a dose of vitamin C in the morning with this vegan green smoothie. Along with kale and avocado, there's a hit of zesty lime and pineapple.","steps":"Put all of the ingredients into a bullet or smoothie maker, add a large splash of water and blitz. Add more water until you have the desired consistency.","source":"https://www.bbcgoodfood.com/recipes/kale-smoothie","created_at":"2018-06-09T00:22:22.993Z","updated_at":"2018-06-09T00:22:22.993Z","ingredients":[{"id":7,"description":"2 handfuls kale"},{"id":8,"description":"½ avocado"},{"id":9,"description":"½ lime, juice only"},{"id":10,"description":"large handful frozen pineapple chunks"},{"id":11,"description":"medium-sized chunk ginger"},{"id":12,"description":"1 tbsp cashew nuts"},{"id":13,"description":"1 banana, optional"}]}
]

export default drinks;