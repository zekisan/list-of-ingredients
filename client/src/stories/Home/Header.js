import React from 'react';

import { storiesOf } from '@storybook/react';
import Header from '../../components/presentational/Home/Header';

storiesOf('Header', module).add('to Storybook', () => <Header />);