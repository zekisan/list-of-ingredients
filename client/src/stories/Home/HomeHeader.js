import React from 'react';

import { storiesOf } from '@storybook/react';
import HomeHeader from '../../HomeHeader';

storiesOf('HomeHeader', module).add('to Storybook', () => <HomeHeader />);