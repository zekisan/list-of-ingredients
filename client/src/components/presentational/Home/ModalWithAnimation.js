import React from 'react';
import { Button, Header, Image, Modal, Divider } from 'semantic-ui-react';

import AnimatedBox from './AnimatedBox';

const ModalWithAnimation = () => (
  <Modal trigger={<Button>Show Modal</Button>} centered={false}>
    <Modal.Header>Select a Photo</Modal.Header>
    <Modal.Content image>
      <Image wrapped size='medium' src='/assets/images/avatar/large/rachel.png' />
      <Modal.Description>
        <Header>Default Profile Image</Header>
        <p>We've found the following gravatar image associated with your e-mail address.</p>
        <p>Is it okay to use this photo?</p>
      </Modal.Description>
    </Modal.Content>
    <Divider section />
    <div>
      <AnimatedBox />
      <Divider section />
      <Button>Click Here</Button>
    </div>
  </Modal>
)

export default ModalWithAnimation;