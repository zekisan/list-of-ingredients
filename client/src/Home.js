import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './App.css';
import { Container, Header, Segment, Button, Dimmer, Loader, Divider } from 'semantic-ui-react';

import HomeHeader from './HomeHeader';

const DrinksGroup = ({ drinks, drink, getDrink }) =>
  <span>
    {
      drinks && drinks.length
        ? <Button.Group color='teal' fluid widths={drinks.length}>
          {Object.keys(drinks).map((key) => {
            return <Button active={drink && drink.id === drinks[key].id} key={key} onClick={() => getDrink(drinks[key].id)}>
              {drinks[key].title}
            </Button>
          })}
        </Button.Group>
        : <Container textAlign='center'>No drinks found.</Container>
    }
  </span>;

DrinksGroup.propTypes = {
  drinks: PropTypes.array,
  drink: PropTypes.object,
  getDrink: PropTypes.func,
};

const DrinkRecipe = ({ drink }) => <span>
  {drink &&
    <Container>
      <Header as='h2'>{drink.title}</Header>
      {drink.description && <p>{drink.description}</p>}
      {drink.ingredients &&
        <Segment.Group>
          {drink.ingredients.map((ingredient, i) => <Segment key={i}>{ingredient.description}</Segment>)}
        </Segment.Group>
      }
      {drink.steps && <p>{drink.steps}</p>}
      {drink.source && <Button basic size='tiny' color='teal' href={drink.source}>Source</Button>}
    </Container>
  }
</span>;

DrinkRecipe.propTypes = {
  drink: PropTypes.object,
};

class Home extends Component {
  constructor() {
    super();

    this.state = {}
    this.getDrinks = this.getDrinks.bind(this);
    this.getDrink = this.getDrink.bind(this);
  }

  componentDidMount() {
    this.getDrinks();
  }

  fetch(endpoint) {
    return window.fetch(endpoint)
      .then(response => response.json())
      .catch(error => console.log(error));
  }

  getDrinks() {
    this.fetch('/api/drinks')
      .then(drinks => {
        if (drinks.length) {
          this.setState({ drinks: drinks })
          this.getDrink(drinks[0].id)
        } else {
          this.setState({ drinks: [] })
        }
      })
  }

  getDrink(id) {
    this.fetch(`/api/drinks/${id}`)
      .then(drink => this.setState({ drink: drink }))
  }

  render() {
    let { drinks, drink } = this.state;
    return drinks
      ? <Container text>
        <HomeHeader />
        <Divider hidden section />
        <DrinksGroup drinks={drinks} drink={drink} getDrink={this.getDrink} />
        <Divider section />
        <DrinkRecipe drink={drink}/>
      </Container>
      : <Container text>
        <Dimmer active inverted>
          <Loader content='loading' />
        </Dimmer>
      </Container>
  }
}

export default Home;
