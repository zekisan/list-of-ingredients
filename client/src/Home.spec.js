import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Home from './Home';
import drinks from './mocks/drinks'

describe('testing Home', () => {
    beforeEach(() => {
        fetch.resetMocks()
        fetch.mockResponseOnce(JSON.stringify({ drinks: drinks }))
    })

    it('renders without crashing', () => {
        shallow(<Home />);
    });

    it('tries snapshots', () => {
        const tree = renderer
            .create(<Home />)
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });
})