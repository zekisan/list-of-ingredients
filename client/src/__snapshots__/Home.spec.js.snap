// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Home tries snapshots 1`] = `
<div
  className="ui text container"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="ui text loader"
      >
        loadingsss
      </div>
    </div>
  </div>
</div>
`;
